pkg/controller/job/job_controller.go

```diff
@@ -795,7 +796,11 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (forget bool, rEr
 		deleted, err := jm.deleteActivePods(ctx, &job, activePods)
 		if uncounted == nil {
 			// Legacy behavior: pretend all active pods were successfully removed.
-			deleted = active
+			if strings.Contains(finishedCondition.Message, "has condition for pending") {
```

Comment:
this is the legacy behavior when there is no tracking with finalizers... the plan is to remove this code in 1.27.

Reply:
So I notice that this code https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/job/job_controller_test.go#L2010 basically doesn't set the finalizers path to true at all.  I think it is set with 1 case and the rest may be default to false.  

So when I was debugging the integration tests I was hitting this path quite a bit.  Maybe we should look at those tests and make sure they are testing the finalizer route?

Reply:
we test both paths until we remove the code https://github.com/kubernetes/kubernetes/blob/a6d66d15f691cb6473f1361caed08e5bd7eae68b/pkg/controller/job/job_controller_test.go#L3019

Reply:
I see now!